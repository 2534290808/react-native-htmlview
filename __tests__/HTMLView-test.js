var React = require('react-native')
var TestUtils = require('react-addons-test-utils')
var HTMLView = require('../HTMLView')
jest.setMock('../htmlToElement', jest.fn())
var htmlToElement = require('../htmlToElement')

describe('HTMLView', () => {
  it('renders the html', () => {
    var htmlContent = '<p><a href="http://jsdf.co">&hearts; nice job!</a></p>'

    htmlToElement.mockImplementation((value, opts, callback) => {
      callback(null, <Text />)
    })

    var shallowRenderer = TestUtils.createRenderer()

    shallowRenderer.render(
      <HTMLView
        value={htmlContent}
      />
    )

    var first = shallowRenderer.getRenderOutput()
    expect(first.props.children).toEqual(undefined)

    shallowRenderer._instance._instance.componentDidMount()

    var second = shallowRenderer.getRenderOutput()
    expect(second.props.children[0].type).toBe(React.Text)
  })
})
