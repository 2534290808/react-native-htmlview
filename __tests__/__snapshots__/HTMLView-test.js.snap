// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HTMLView/> can use a custom node class 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
          
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={null}
    selectable={false}
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
              
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={null}
      selectable={false}
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
              
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
          
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> can use a custom renderer 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
          
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={null}
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
              
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
        
  </Text>
</View>
`;

exports[`<HTMLView/> can use custom node props 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
          
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={null}
    selectable={false}
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
              
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={null}
      selectable={false}
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
              
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
          
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> should not render extra linebreaks if configured not to 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
          
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={null}
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
                
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={null}
      style={
        Object {
          "fontSize": 30,
          "fontWeight": "500",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Dwayne’s only companion at night was a Labrador retriever named Sparky.
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
              
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={null}
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
                Sparky could not wag his tail-because of an automobile accident many years ago, so he had no way of telling other dogs how friendly he was.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={null}
        style={null}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
                He opened the door of the cage, something Bill couldn’t have done in a thousand years. Bill flew over to a windowsill.
                
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
            
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
        
  </Text>
</View>
`;

exports[`<HTMLView/> should render an <Image /> with set width/height 1`] = `
<View
  style={undefined}
>
  <Image
    source={
      Object {
        "height": 58,
        "uri": "https://facebook.github.io/react-native/img/header_logo.png",
        "width": 66,
      }
    }
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 58,
        "width": 66,
      }
    }
  />
</View>
`;

exports[`<HTMLView/> should render an <Image />, with default width/height of 1 1`] = `
<View
  style={undefined}
>
  <Image
    source={
      Object {
        "height": 1,
        "uri": "https://facebook.github.io/react-native/img/header_logo.png",
        "width": 1,
      }
    }
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 1,
        "width": 1,
      }
    }
  />
</View>
`;

exports[`<HTMLView/> should render an empty <Text/> element 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={null}
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "#007AFF",
          "fontWeight": "500",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        &hearts nice job!
      </Text>
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> should render ol numbers 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={null}
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={null}
      style={null}
    >
      1. 
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         a 
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={null}
      style={null}
    >
      2. 
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         b 
      </Text>
    </Text>
  </Text>
</View>
`;

exports[`<HTMLView/> should render shoddy html including headings, links, bold, italic 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
          
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={null}
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
              
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={null}
      style={null}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
                  
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={null}
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "500",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          &gt; Dwayne’s only companion at night was a Labrador retriever named Sparky.
        </Text>
        
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
                
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={null}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          
                  
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={null}
          style={
            Object {
              "fontStyle": "italic",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Sparky could not wag his tail-because of an automobile accident many years ago, so he had no way of telling other dogs how friendly he was.
                    He opened the door of the cage, something Bill couldn’t have done in a thousand years. Bill flew over to a windowsill.
                    
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={null}
            style={
              Object {
                "fontWeight": "500",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              The undippable flag was a beauty, and the anthem and the vacant motto might not have mattered much, if it weren’t for this: a lot of citizens were so ignored and cheated and insulted that they thought they might be in the wrong country, or even on the wrong planet, that some terrible mistake had been made.
                      
            </Text>
          </Text>
        </Text>
        
        
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
                
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={null}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          
                    [1] 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "color": "#007AFF",
              "fontWeight": "500",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            https://code.facebook.com/posts/1505962329687926/flow-a-new-...
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          
                  
        </Text>
        
        
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        
              
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      
          
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
        
  </Text>
</View>
`;

exports[`<HTMLView/> should render text in a (nested) <Text/> element 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    This is some text
  </Text>
</View>
`;

exports[`<HTMLView/> should render ul bullets 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={null}
    style={null}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={null}
      style={null}
    >
      • 
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         a 
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={null}
      style={null}
    >
      • 
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
         b 
      </Text>
    </Text>
  </Text>
</View>
`;
